signer/src/block_observer.rs

```diff
@@ -472,6 +478,34 @@ impl<C: Context, B> BlockObserver<C, B> {
 
         Ok(())
     }
+
+    /// Update the sBTC peg limits from Emily
+    async fn update_sbtc_limits(&self) -> Result<(), Error> {
+        let mut limits = self.context.get_emily_client().get_limits().await?;
+        let max_mintable = match limits.total_cap() {
+            None => Amount::MAX_MONEY,
+            Some(total_cap) => {
+                let sbtc_supply = self
+                    .context
+                    .get_stacks_client()
+                    .get_sbtc_total_supply(&self.context.config().signer.deployer)
+                    .await?;
+                // The maximum amount of sBTC that can be minted is the total cap
+                // minus the current supply.
+                total_cap.checked_sub(sbtc_supply).unwrap_or(Amount::ZERO)
+            }
+        };
+        limits.set_max_mintable_cap(Some(max_mintable));
+
+        let signer_state = self.context.state();
+        if limits == signer_state.get_current_limits() {
+            tracing::trace!(%limits, "sBTC limits have not changed");
+        } else {
+            tracing::debug!(%limits, "updated sBTC limits from Emily");
+            signer_state.update_current_limits(limits);
```

Comment:
If you constructed a new `SbtcLimits` struct here then you wouldn't need the `set_max_mintable_cap()` method (see earlier comment)

Reply:
done in https://github.com/stacks-network/sbtc/pull/811/commits/086c1a9a380d8416c63ea1edca2d3307453a0bf1